IDE/ECLIPSE/DEOS/deos_malloc.c

```diff
@@ -0,0 +1,94 @@
+/* deos_malloc.c
+ *
+ * Copyright (C) 2018 wolfSSL Inc.
+ *
+ * This file is part of wolfSSL.
+ *
+ * wolfSSL is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * wolfSSL is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1335, USA
+ */
+#include <wolfssl/wolfcrypt/settings.h>
+#include <wolfssl/ssl.h>
+
+#define HEAP_SIZE_MAX   (1*1024*1024)
+
+static size_t allocatedMemory = 0;
+
+size_t getMemAllocatedSize_deos(size_t* size){
+
+    if (size)
+        *size = allocatedMemory;
+
+    return allocatedMemory;
+}
+
+void free_deos(void *ptr) {
+    //printf("fake free_deos()\n");
+    return;
+}
+
+void *realloc_deos(void *ptr, size_t size) {
+    void *newptr;
+
+    if (size == 0)
+        return ptr;
+
+    newptr = malloc_deos(size);
+    free_deos(ptr);
```

Comment:
realloc() is supposed to copy the data from the old ptr into the new space.

Reply:
Good catch! I left out XMEMCPY intentionally. I can include a description that the caller is responsible for copying the old data.  Copying the data can only be done by the caller since realloc_deos() can't access the size of the old buffer.  I need to look again in ssl.c but there is always a call to XMEMCPY after XREALLOC calls.  Any thoughts?

Reply:
That doesn't work.  The caller is provided a contract that the new allocation succeeds with the copy or not at all.  The caller has no way to copy the old data since it's no longer accessible.  We can't have a "special" realloc that doesn't work like realloc push off responsibility.